Bio/PDB/MMCIFParser.py

```diff
@@ -67,6 +67,49 @@ def get_structure(self, structure_id, filename):
 
     # Private methods
 
+    def _mmcif_get(self, key, dict, deflt):
+        if key in dict:
+            rslt = dict[key][0]
+            if "?" != rslt:
+                return rslt
+        return deflt
+
+    def _get_mmcif_header_dict(self, md):
+        dict = {
+            "name": "",
+            "head": "",
+            "idcode": "",
+            "deposition_date": "",
+            "structure_method": "",
+            "resolution": 0.0,
+        }
+
+        dict["idcode"] = self._mmcif_get("_entry.id", md, None)
```

Comment:
Would it be reasonable to refactor the `__mmcif_get` function to take a list of keys, in order of preference, and keep trying until finding a value (or returning the default at the end)? It would avoid all these `if` statements here and more extensible?

Reply:
done.